### ISSUE TITLE
LG G3?

### ISSUE THREAD

**Issue by PseudoDistant** [msm8974-mainline/linux#16]
The G3 ran a MSM8974AC (Snapdragon 801) chipset, would it be supported?
The G2 also ran a MSM8974 (Snapdragon 800), would it be supported as well?

**Comment by z3ntu** [comment]
Hi! If you have the device, it's not too difficult to write a devicetree file for that device to get it booting mainline Linux. But I don't own one of these devices so it's something for you to do.

**Comment by PseudoDistant** [comment]
> Hi! If you have the device, it's not too difficult to write a devicetree file for that device to get it booting mainline Linux. But I don't own one of these devices so it's something for you to do.

Would I have to compile AOSP? I don't have enough RAM to compile Android yet, still waiting on a good deal for a 128 GB kit. (I know 64 GB is enough, but 32 GB was enough very recently and now it doesn't work anymore.)

Would a mainline Linux kernel work with Android 11? I could try it if it'll work.

**Comment by z3ntu** [comment]
While it is possible to build and run Android with mainline kernel, it's not done often so doing that you'd need to do a lot of Android adaption work yourself.

But normally people working on developing mainline kernel run an OS like postmarketOS or Mobian. And also since you don't seem to be very familiar with this topic, mainline kernel still has a good number of missing features for device adaption for MSM8974, so e.g. audio is not working on any msm8974 device yet so without significant time investment you won't get a fully functional device anytime soon.

**Comment by PseudoDistant** [comment]
You are correct in your assessment on my acquaintance with this topic, but it is one I'm trying to learn. I have a few different older LG phones, and am looking for ways to keep them updated. I've been researching this topic for a while now...

As far as I'm aware, the GKI project seems to be the most likely avenue of success, so long as the ACK patches can be merged in without too many issues, not that I'm asking you to do this, I just want to know if I understand this right. I understand the value in mobile Linux and have been fairly curious about postmarketOS and Plasma Mobile myself... 

As for setting that up, would I have to compile postmarketOS myself or would just the kernel be enough if there's a 32-bit armv7 image available?

**Comment by z3ntu** [comment]
For postmarketOS all packages are provided as binary either by Alpine Linux or postmarketOS repositories, so a "build" should take barely any resources, just a bit of bandwidth to download the packages.

But for compiling the kernel from source - which you'd need to do for development, doing the device bringup in kernel - an initial build could take 30 minutes or so depending on your PC. After that incremental builds should be way faster, depending how you build only a couple of seconds.
